### PR TITLE
Add an UseRouter that takes Action<IRouteBuilder>

### PR DESCRIPTION
**Pull request by Muchiachio**
- Added an overload of UseRouter that takes Action<IRouteBuilder>, to make configuring standalone routing much less verbose.

Addresses #332

### CODE REVIEW

File: src/Microsoft.AspNetCore.Routing/BuilderExtensions.cs [hunk 1]
```diff
@@ -41,5 +41,30 @@ public static IApplicationBuilder UseRouter(this IApplicationBuilder builder, IR
...
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            if (action == null)
+            {
+                throw new ArgumentNullException(nameof(action));
+            }
+
+            IRouteBuilder routeBuilder = new RouteBuilder(builder);
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Routing/BuilderExtensions.cs]: use `var` here

**Comment by rynowak** [comment]
Would be nice to see a test of this in action, even if it's not really a "unit test"

**Comment by Muchiachio** [comment]
Like the ones in sample pages? Because I tried adding unit tests, but that configuration buried behind a lot of layers :smiley:

**Comment by rynowak** [comment]
It shouldn't be hard to do using something like https://github.com/aspnet/Routing/blob/dev/test/Microsoft.AspNetCore.Routing.Tests/RequestDelegateRouteBuilderExtensionsTest.cs as a starting point. If this makes you barf let me know, I don't mind doing it as part of the merge.

**Comment by Muchiachio** [comment]
Yes that would be great, sorry I couldn't add them myself :disappointed_relieved:

**Comment by rynowak** [comment]
No problem, I'll take care of it.

**Comment by rynowak** [comment]
4adc693cb52c3b70175721ed9c474cde46409f96 and 0978a7dd166472f30ac40138fcc71da4cfb1df61 merged. Thanks :+1:

**Pull request closed without merging**